### PR TITLE
Improve allowFailBuild option visibility and help section

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-azure-devops-extension#testing) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] This pull request is on the dev branch.
- [ ] I used `npm run format` for formatting the code before submitting the pull request.
-----
following https://github.com/jfrog/jfrog-azure-devops-extension/issues/386

### CODE REVIEW

File: tests/tests.ts [hunk 1]
```diff
@@ -963,7 +962,7 @@ function runSyncTest(description: string, testFunc: () => void, skip?: boolean):
     it(description, (done): void => {
         testFunc();
         done();
-    }).timeout(300000); // 5 minutes
+    }).timeout(400000); // 5 minutes
```
Comment (yahavi) [inline tests/tests.ts]: It is acually 6.6666667 minutes :)

**Review by yahavi — CHANGES_REQUESTED** [review]

File: tests/tests.ts (hunk 1 above)
Comment (sverdlov93) [reply tests/tests.ts]: changed to 360000

File: tasks/JFrogBuildScan/task.json [hunk 2]
```diff
@@ -57,15 +57,15 @@
             "label": "Allow fail build",
             "defaultValue": "true",
             "required": true,
-            "helpMarkDown": "Uncheck if you do not wish to fail the build, following rules with 'Fail Build' actions."
+            "helpMarkDown": "Uncheck if you do not wish the command to return exit code 3, even if found a security violation that matches an Xray policy with 'Fail Build' rule."
```
Comment (eyalbe4) [inline tasks/JFrogBuildScan/task.json]: Uncheck if you do not wish the command to return exit code 3, even if found a security violation that matches an Xray policy with 'Fail Build' rule.
-->
Unset if you don't wish the audit process to exit with code 3, if a violation matching an Xray policy is found, and the policy includes a 'Fail Build' rule.

Let's make this change in the other two `task.json` files in this PR.

File: tasks/JFrogAudit/task.json [hunk 3]
```diff
@@ -94,7 +94,8 @@
             "label": "Allow fail build",
             "defaultValue": "true",
             "required": true,
-            "helpMarkDown": "Uncheck if you do not wish to fail the build, following rules with 'Fail Build' actions."
+            "visibleRule": "watchesSource!=none",
```
Comment (eyalbe4) [inline tasks/JFrogAudit/task.json:97]: Shouldn't this condition include `project` as well? A policy which includes `Fail build` can also be associated with a Project and not just a Watch.

Also notice that this visibility condition should be modified in one of the other two `task.json` files. One of the `task.json` is missing a visibility condition altogether.

File: tasks/JFrogBuildScan/task.json [hunk 4]
```diff
@@ -57,15 +57,15 @@
...
         },
         {
             "name": "vuln",
             "type": "boolean",
             "label": "Show all vulnerabilities",
             "defaultValue": "false",
             "required": true,
-            "helpMarkDown": "Set to true if you'd like to receive all vulnerabilities, regardless of the policy configured in Xray. (This functionality requires version 3.40.0 or above of JFrog Xray)"
+            "helpMarkDown": "Set to true if you'd like to receive an additional view of all vulnerabilities, regardless of the policy configured in Xray. (This functionality requires version 3.40.0 or above of JFrog Xray)"
```
Comment (eyalbe4) [inline tasks/JFrogBuildScan/task.json]: Set to true if you'd like to receive an additional view of all vulnerabilities, regardless of the policy configured in Xray. (This functionality requires version 3.40.0 or above of JFrog Xray)
-->
Set to true if you'd like to see an additional view with all vulnerabilities, regardless of the policy configured in Xray. (This functionality requires version 3.40.0 or above of JFrog Xray)

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: tasks/JFrogAudit/task.json (hunk 3 above)
Comment (sverdlov93) [reply tasks/JFrogAudit/task.json:97]: @eyalbe4 
1. watchesSource includes watches/repopath/projects 
2. it's on purpose, the build-scan task is a context command by design. it's always connected to a watch

**Pull request merged**